recipes-core/initrdscripts/files/ni_provisioning.safemode

```diff
@@ -319,14 +295,8 @@ install_grubenv()
 	grub-editenv $BOOTFS_MOUNTPOINT/grub/grubenv set "BIOSBootMode=efi"
 	grub-editenv $BOOTFS_MOUNTPOINT/grub/grubenv set "NITarget=$NI_TARGET"
 
-	# If target has a TPM chip or runs on Hyper-V, we need to enable an alternate boot flow
-	# involving initramfs and bootflags dir
-	if $use_ramfs; then
-		grub-editenv $BOOTFS_MOUNTPOINT/grub/grubenv set "enable_initramfs=1"
```

Comment:
So going forward, a target provisioned with old safemode won't be able to boot new runmode on targets with TPM chip. I guess this is acceptable to us? Should we add checks somewhere to ensure new runmode isn't installed on old safemode?

Reply:
Sort of, old safemode will still boot on all targets, however if enable_initramfs is set to 1 (because the target has a TPM chip, or whatever), the new runmode won't boot since the initramfs will be "missing on purpose due to this change".  If the old safemode has enable_initramfs==0, then it will boot the old safemode/new runmode combo just fine since neither one requires the initramfs.

Does that answer your question?

Reply:
I understand that thats the case. I'm wondering if/how we will enforce it - right now I believe MAX doesn't stop us from installing hardknott runmode on a sumo safemode. Should we create a story to research/implement this?

Reply:
PS: Have we broken safemode compatibility for new releases like this before? @billpittman, @jeminor

Reply:
I don't believe we've broken compatibility necessarily.  In the case of a TPM enabled system, then yes, but we've dropped that capability, so its a compatibility break in the sense that we don't support it.  Safemode still installs runmode, even an older safemode.

Reply:
@chaitu236 Pointed out a specific case where TPM capable targets would require a safemode upgrade in order to properly install.  Without the update, MAX would "successfully" provision the target, but when it boots into the newly installed runmode, it would fail to find the initramfs file and force the target into  safemode with a software installation error.

Do we have a way to "force" a safemode upgrade from MAX or inform the user that the "problem" is the old safemode?  We used to have a mechanism like that, but it was removed a very long time ago and I don't remember if we put in something else in its place.

If we don't have a way to inform the user that there's a safemode upgrade required, we may have to delay/abandon this change for the moment until we can handle TPM capable systems.

Reply:
I think we do have a way for MAX to notify customers that a safemode upgrade is required. Could you ask Dan and co.

Reply:
There is a mechanism and can require it be invoked if needed.  Specifically, we can require that a system image install require >= some specific safemode.

I'm fine with this, and we can invoke that mechanism when we need it.

Reply:
@billpittman can you create a story parented to hardknott work so we don't forget it.